shared/workspace_repo/workspace.go

```diff
@@ -311,6 +311,10 @@ func (s *TFEWorkspace) WaitWithOptions(ctx context.Context, waitOptions options.
 		tfe.RunPlannedAndFinished: {},
 	}
 
+	dryRun, ok := ctx.Value(options.DryRunKey).(bool)
+	if !ok {
+		dryRun = false
+	}
 	if dryRun {
 		RunDoneStatuses = map[tfe.RunStatus]bool{
```

Comment:
The first run that creates the workspace doesn't run in a dryrun mode. To address it, tfe.RunApplied should also be accepted as `RunDoneStatuses` here.

Reply:
Also, TfeSuccessStatuses should include tfe.RunApplied for the same reason.